src/main/java/org/jabref/gui/externalfiles/FindFullTextAction.java

```diff
@@ -71,25 +73,34 @@ public void execute() {
         }
 
         Task<Map<BibEntry, Optional<URL>>> findFullTextsTask = new Task<Map<BibEntry, Optional<URL>>>() {
+
             @Override
             protected Map<BibEntry, Optional<URL>> call() {
                 Map<BibEntry, Optional<URL>> downloads = new ConcurrentHashMap<>();
                 int count = 0;
-                for (BibEntry entry : basePanel.getSelectedEntries()) {
-                    FulltextFetchers fetchers = new FulltextFetchers(Globals.prefs.getImportFormatPreferences());
-                    downloads.put(entry, fetchers.findFullTextPDF(entry));
+                AutoSetFileLinksUtil util = new AutoSetFileLinksUtil(basePanel.getBibDatabaseContext(), Globals.prefs.getFilePreferences(), Globals.prefs.getAutoLinkPreferences(), ExternalFileTypes.getInstance());
+                List<BibEntry> changed = util.linkAssociatedFiles(basePanel.getSelectedEntries(), new NamedCompound(""));
+
+                for (BibEntry entry : changed) {
+                    if (entry.getFiles().isEmpty()) {
+                        FulltextFetchers fetchers = new FulltextFetchers(Globals.prefs.getImportFormatPreferences());
```

Comment:
Would it be possible to instantiate this before the `for` loop? It seems that this variable could be reused (because it's fetcher**s**).